connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -169,6 +172,7 @@ public String version() {
         return listConsumerGroupOffsets(group).entrySet().stream()
             .filter(x -> shouldCheckpointTopic(x.getKey().topic()))
             .map(x -> checkpoint(group, x.getKey(), x.getValue()))
+            .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty)) // do not emit checkpoints for partitions that don't have offset-syncs
```

Comment:
There is a "feature" in checkpointing which copies the offsets of replica topics back to target as-is. (I'm using quotes because the copy doesn't really make any sense, the offsets should be properly translated.) E.g. there are clusters A and B, replication enabled in both direction. Topic "test" is replicated A->B, which creates "A.topic" in B. If the topic filter matches "A.topic", the topic itself won't be replicated, but checkpointing can still pick it up (there is no cycle detection in the shouldCheckpointTopic method). In its current state, checkpointing will handle this as a special case, and will copy the committed offset of "A.topic" to A, but without the prefix (there is also a bug in the implementation, which strips all prefixes instead of the last one).
I think this feature is not useful at all (and also has a bug). My question is, whether that feature will be dropped, or somehow handles in this change as well. "A.topic" does not have an offset mapping in the offset-syncs topic, as it is never replicated back to A - this flatMap will filter all such topics.
I think this should be handled separately, or the feature should be removed completely - in that case the MirrorCheckpointTask.renameTopicPartition method should be updated accordingly.

Reply:
Is it possible to skip these topics so that the half-finished feature stays intact that so we can finish it later?

Reply:
Thanks @urbandan and @viktorsomogyi. I think I vaguely understand what you mean, I plan to take a look at the code again this afternoon so hopefully that will become clearer. 

I doubt this "feature" was by design (maybe @ryannedolan can shed some light) as it does not seem useful or it seems pretty broken. Considering we're late in the release process of 3.2, I'll see if I can find a heuristic to filter these topics and keep the current behavior.

Reply:
@viktorsomogyi mentioned you also looked at this issue. Apart from the weird "feature" described, how different is it from your fix? Is there another approach you'd favor or is this correctly addressing the issue?

Reply:
I've looked at this more closely. 

MirrorSourceConnector correctly prevents loops but it's not the case for MirrorCheckpointConnector. It's not clear why they have different `shouldReplicateTopic()` methods. I don't understand the point of this behavior, surely it must be a bug. I don't see a use in having untranslated checkpoints from remote offsets. 

If we agree it's unexpected behavior and can't see why anybody would rely on this, I think it's ok to remove it. Then this would allow us to do some cleanups, ie update `renameTopicPartition()`, have a single `shouldReplicateTopic()` method.

Reply:
@mimaison our fix is pretty much the same but a bit more simplistic (uses -1 instead of Optional). I also think no one would rely on that weird behavior so I'm favor in removing that functionality.

Reply:
I agree that having untranslated offsets from remote topics seems weird. Actually, I was believing that the checkpoint connector was doing the reverse-translation for offsets of remote topics. If it does not then I see it as a big limitation because once a consumer group has moved from a source topic to a remote topic, it will not be able to switch back to the source topic again (smartly resuming from where it stopped). So my question is: will this feature be handled separately?

Reply:
@dadufour I'm facing similar issue w.r.to the consumer group offset handling during the failback and thank you for lifting it up in the Strimzi forum that help me land up in this thread.  @mimaison I thought shall check if you have any idea if there is a KIP to handle this behavior in MM2 or is it already fixed?

Reply:
@sgn2607 I'm not aware of anybody working on this. I don't think there's even a Jira clearly stating this requirement. Feel free to work on it if you want.